### ISSUE TITLE
Cannot play any youtube video

### ISSUE THREAD

**Issue by JuganD** [lavalink-devs/youtube-source#93]
### Basic Troubleshooting

- [X] I have checked for similar issues (open AND closed), and my issue is not a duplicate. (MANDATORY)
- [X] I have checked for pull requests that may already address my issue. (MANDATORY)
- [X] I am using (and my issue is reproducible on) the latest version of youtube-source.
- [X] My issue is reproducible with IPv6 rotation.

### Version of youtube-source

1.11.1

### The search query/queries, URL(s), playlist ID(s)/URL(s) or video ID(s)/URL(s) that triggered the issue

any video really, for example: https://www.youtube.com/watch?v=a3ICNMQW7Ok

### Code Example

N/A

### Exception and Stacktrace

2024-12-15T21:04:34.471Z  INFO 1 --- [Lavalink] [  XNIO-1 task-2] l.server.player.AudioLoaderRestHandler   : Got request to load for identifier "https://www.youtube.com/watch?v=a3ICNMQW7Ok"
2024-12-15T21:04:43.788Z  INFO 1 --- [Lavalink] [  XNIO-1 task-2] l.server.player.AudioLoaderRestHandler   : Loaded track Wildlife Windows 7 Sample Video
2024-12-15T21:04:43.844Z  INFO 1 --- [Lavalink] [  XNIO-1 task-2] l.server.io.RequestLoggingFilter         : GET /v4/loadtracks?identifier=https%3a%2f%2fwww.youtube.com%2fwatch%3fv%3da3ICNMQW7Ok
2024-12-15T21:04:43.953Z  INFO 1 --- [Lavalink] [  XNIO-1 task-2] l.server.io.RequestLoggingFilter         : PATCH /v4/sessions/.....
2024-12-15T21:04:50.158Z ERROR 1 --- [Lavalink] [back-1-thread-1] c.s.d.l.t.p.LocalAudioTrackExecutor      : Error in playback of a3ICNMQW7Ok

com.sedmelluq.discord.lavaplayer.tools.FriendlyException: Something broke when playing the track.
        at com.sedmelluq.discord.lavaplayer.tools.ExceptionTools.wrapUnfriendlyExceptions(ExceptionTools.java:45) ~[lavaplayer-2.2.2.jar!/:na]
        at com.sedmelluq.discord.lavaplayer.track.playback.LocalAudioTrackExecutor.execute(LocalAudioTrackExecutor.java:121) ~[lavaplayer-2.2.2.jar!/:na]
        at com.sedmelluq.discord.lavaplayer.player.DefaultAudioPlayerManager.lambda$executeTrack$2(DefaultAudioPlayerManager.java:339) ~[lavaplayer-2.2.2.jar!/:na]
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[na:na]
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[na:na]
        at java.base/java.lang.Thread.run(Thread.java:840) ~[na:na]
Caused by: java.lang.RuntimeException: java.net.UnknownHostException: rr3---sn-uxaxiv0nxx5q-nv4l.googlevideo.com: Try again
        at com.sedmelluq.discord.lavaplayer.container.matroska.MatroskaAudioTrack.loadMatroskaFile(MatroskaAudioTrack.java:57) ~[lavaplayer-2.2.2.jar!/:na]
        at com.sedmelluq.discord.lavaplayer.container.matroska.MatroskaAudioTrack.process(MatroskaAudioTrack.java:35) ~[lavaplayer-2.2.2.jar!/:na]
        at com.sedmelluq.discord.lavaplayer.track.DelegatedAudioTrack.processDelegate(DelegatedAudioTrack.java:25) ~[lavaplayer-2.2.2.jar!/:na]
        at dev.lavalink.youtube.track.YoutubeAudioTrack.processStatic(YoutubeAudioTrack.java:160) ~[youtube-plugin-1.11.1.jar!/:na]
        at dev.lavalink.youtube.track.YoutubeAudioTrack.processWithClient(YoutubeAudioTrack.java:128) ~[youtube-plugin-1.11.1.jar!/:na]
        at dev.lavalink.youtube.track.YoutubeAudioTrack.process(YoutubeAudioTrack.java:74) ~[youtube-plugin-1.11.1.jar!/:na]
        at com.sedmelluq.discord.lavaplayer.track.playback.LocalAudioTrackExecutor.execute(LocalAudioTrackExecutor.java:109) ~[lavaplayer-2.2.2.jar!/:na]
        ... 4 common frames omitted
        Suppressed: dev.lavalink.youtube.ClientInformation:
  yts.version: 1.11.1
  client.identifier: WEB
  client.options: ClientOptions{playback=true, playlistLoading=true, videoLoading=true, searching=true}
Caused by: java.net.UnknownHostException: rr3---sn-uxaxiv0nxx5q-nv4l.googlevideo.com: Try again
        at java.base/java.net.Inet6AddressImpl.lookupAllHostAddr(Native Method) ~[na:na]
        at java.base/java.net.InetAddress$PlatformNameService.lookupAllHostAddr(InetAddress.java:934) ~[na:na]
        at java.base/java.net.InetAddress.getAddressesFromNameService(InetAddress.java:1543) ~[na:na]
        at java.base/java.net.InetAddress$NameServiceAddresses.get(InetAddress.java:852) ~[na:na]
        at java.base/java.net.InetAddress.getAllByName0(InetAddress.java:1533) ~[na:na]
        at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1385) ~[na:na]
        at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1306) ~[na:na]
        at org.apache.http.impl.conn.SystemDefaultDnsResolver.resolve(SystemDefaultDnsResolver.java:45) ~[httpclient-4.5.14.jar!/:4.5.14]
        at com.sedmelluq.discord.lavaplayer.tools.http.ExtendedConnectionOperator.resolveAddresses(ExtendedConnectionOperator.java:126) ~[lavaplayer-2.2.2.jar!/:na]
        at com.sedmelluq.discord.lavaplayer.tools.http.ExtendedConnectionOperator.connect(ExtendedConnectionOperator.java:57) ~[lavaplayer-2.2.2.jar!/:na]
        at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:376) ~[httpclient-4.5.14.jar!/:4.5.14]
        at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:393) ~[httpclient-4.5.14.jar!/:4.5.14]
        at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:236) ~[httpclient-4.5.14.jar!/:4.5.14]
        at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:186) ~[httpclient-4.5.14.jar!/:4.5.14]
        at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89) ~[httpclient-4.5.14.jar!/:4.5.14]
        at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110) ~[httpclient-4.5.14.jar!/:4.5.14]
        at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185) ~[httpclient-4.5.14.jar!/:4.5.14]
        at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83) ~[httpclient-4.5.14.jar!/:4.5.14]
        at com.sedmelluq.discord.lavaplayer.tools.io.HttpInterface.execute(HttpInterface.java:73) ~[lavaplayer-2.2.2.jar!/:na]
        at com.sedmelluq.discord.lavaplayer.tools.io.PersistentHttpStream.attemptConnect(PersistentHttpStream.java:121) ~[lavaplayer-2.2.2.jar!/:na]
        at com.sedmelluq.discord.lavaplayer.tools.io.PersistentHttpStream.connect(PersistentHttpStream.java:106) ~[lavaplayer-2.2.2.jar!/:na]
        at com.sedmelluq.discord.lavaplayer.tools.io.PersistentHttpStream.internalRead(PersistentHttpStream.java:158) ~[lavaplayer-2.2.2.jar!/:na]
        at com.sedmelluq.discord.lavaplayer.tools.io.PersistentHttpStream.read(PersistentHttpStream.java:174) ~[lavaplayer-2.2.2.jar!/:na]
        at java.base/java.io.DataInputStream.readUnsignedByte(DataInputStream.java:288) ~[na:na]
        at java.base/java.io.DataInputStream.readByte(DataInputStream.java:268) ~[na:na]
        at com.sedmelluq.discord.lavaplayer.container.matroska.format.MatroskaEbmlReader.readEbmlInteger(MatroskaEbmlReader.java:39) ~[lavaplayer-2.2.2.jar!/:na]
        at com.sedmelluq.discord.lavaplayer.container.matroska.format.MatroskaFileReader.readNextElement(MatroskaFileReader.java:47) ~[lavaplayer-2.2.2.jar!/:na]
        at com.sedmelluq.discord.lavaplayer.container.matroska.MatroskaStreamingFile.readFile(MatroskaStreamingFile.java:94) ~[lavaplayer-2.2.2.jar!/:na]
        at com.sedmelluq.discord.lavaplayer.container.matroska.MatroskaAudioTrack.loadMatroskaFile(MatroskaAudioTrack.java:52) ~[lavaplayer-2.2.2.jar!/:na]
        ... 10 common frames omitted

### What is your client configuration?

allowSearch: true, allowDirectVideoIds: false, allowDirectPlaylistIds: false, pot: token + visitorData (tried without them, the same result), clients: WEB, MUSIC, WEBEMBEDDED. My IP doesn't seem to be blocked by YT, I'm playing without any issues using any browser.

**Comment by devoxin** [comment]
> UnknownHostException

This is caused by faulty routing of your device's network. Nothing we can fix here.

**Comment by JuganD** [comment]
Ah thank you kind sir! I didn't thought to check the name resolution from inside the container, I only did it outside, where it works.

For anyone perhaps reading this in the future, this is how you fix it if you run lavalink in a container with docker compose:
1. Use `docker exec -it {container_name_or_id} /bin/sh` to hook into the container
2. Use `cat /etc/resolv.conf` and copy the content
3. Create a `resolv.conf` file next to your docker compose or wherever you like and paste the content
4. Edit the file and add `nameserver {whatever_you_like}`, I added google, so `nameserver 8.8.8.8` after the already existing nameserver, but don't delete the existing one, or your docker networking will suffer
5. Mount the `resolv.conf` to replace the one inside the container
```
volumes:
      - ./resolv.conf:/etc/resolv.conf
```

It is a cheat of sorts, it's not fixing the main DNS issue in all docker, but I fear that might be caused, because I have PiHole as well and it's operating on another docker network than the one of Lavalink, so the DNS resolution fails, because it can't reach PiHole in reality. I don't want to add it to the other network as I have other stuff going on there, so this is a solution for now.
